src/browser/BrowserService.cpp

```diff
@@ -391,14 +391,42 @@ QString BrowserService::storeKey(const QString& key)
     return id;
 }
 
-QString BrowserService::getKey(const QString& id)
+QPair<bool, QString> BrowserService::getKey(const QString& id)
 {
-    auto db = getDatabase();
-    if (!db) {
+    auto currentDatabase = getDatabase();
+    if (!currentDatabase) {
         return {};
     }
 
-    return db->metadata()->customData()->value(CustomData::BrowserKeyPrefix + id);
+    const auto currentHash = currentDatabase->metadata()->customData()->value(CustomData::BrowserKeyPrefix + id);
+    const auto isAssociated = !currentHash.isEmpty();
+    auto result = qMakePair(isAssociated, currentHash);
+
+    // Return the result from current database
+    if (!browserSettings()->searchInAllDatabases()) {
+        return result;
+    }
+
+    // Get all open databases
+    QList<QSharedPointer<Database>> databases;
+    for (auto dbWidget : getMainWindow()->getOpenDatabases()) {
+        auto db = dbWidget->database();
+        if (db && !dbWidget->isLocked()) {
+            databases << db;
+        }
+    }
+
+    // Return the matching hash for current client id
+    for (const auto& db : databases) {
+        for (const QString& key : db->metadata()->customData()->keys()) {
+            if (key.startsWith(CustomData::BrowserKeyPrefix) && key == CustomData::BrowserKeyPrefix + id) {
+                result.second = db->metadata()->customData()->value(key);
+                return result;
+            }
+        }
+    }
```

Comment:
Combine these two for loops, if the database is unlocked then do the key search. You are also re-using the result of `isAssociated` in this block which I think is not appropriate?